lib/awful/layout/init.lua

```diff
@@ -162,6 +162,11 @@ function layout.arrange(screen)
     delayed_arrange[screen] = true
 
     timer.delayed_call(function()
+        if not screen.valid then
```

Comment:
Hu? `.valid` was added specifically so that no `pcall`-related hacks are needed. It exists on all C-API objects and even works on objects that otherwise throw errors when you access them. So, no, this doesn't need a `pcall()` and whatever you mean with "like the clients do" doesn't need a `pcall()` either. What are you referring to?